ext/gd/config.m4

```diff
@@ -108,92 +111,78 @@ dnl
-                 libgd/gd_bmp.c"
+if test "$PHP_GD" != "no"; then
+
+  if test "$PHP_EXTERNAL_GD" = "no"; then
+    GD_MODULE_TYPE=builtin
+    GD_CFLAGS=""
+    extra_sources="libgd/gd.c libgd/gd_gd.c libgd/gd_gd2.c libgd/gd_io.c libgd/gd_io_dp.c \
+                  libgd/gd_io_file.c libgd/gd_ss.c libgd/gd_io_ss.c libgd/gd_webp.c \
+                  libgd/gd_png.c libgd/gd_jpeg.c libgd/gdxpm.c libgd/gdfontt.c libgd/gdfonts.c \
+                  libgd/gdfontmb.c libgd/gdfontl.c libgd/gdfontg.c libgd/gdtables.c libgd/gdft.c \
+                  libgd/gdcache.c libgd/gdkanji.c libgd/wbmp.c libgd/gd_wbmp.c libgd/gdhelpers.c \
+                  libgd/gd_topal.c libgd/gd_gif_in.c libgd/gd_xbm.c libgd/gd_gif_out.c libgd/gd_security.c \
+                  libgd/gd_filter.c libgd/gd_pixelate.c libgd/gd_rotate.c libgd/gd_color_match.c \
+                  libgd/gd_transform.c libgd/gd_crop.c libgd/gd_interpolation.c libgd/gd_matrix.c \
+                  libgd/gd_bmp.c"
 
 dnl check for fabsf and floorf which are available since C99
-  AC_CHECK_FUNCS(fabsf floorf)
+    AC_CHECK_FUNCS(fabsf floorf)
 
 dnl These are always available with bundled library
-  AC_DEFINE(HAVE_GD_BUNDLED,          1, [ ])
-  AC_DEFINE(HAVE_GD_PNG,              1, [ ])
-  AC_DEFINE(HAVE_GD_BMP,              1, [ ])
+    AC_DEFINE(HAVE_GD_BUNDLED,          1, [ ])
+    AC_DEFINE(HAVE_GD_PNG,              1, [ ])
+    AC_DEFINE(HAVE_GD_BMP,              1, [ ])
 
 dnl Various checks for GD features
-  PHP_GD_ZLIB
-  PHP_GD_PNG
-  PHP_GD_WEBP
-  PHP_GD_JPEG
-  PHP_GD_XPM
-  PHP_GD_FREETYPE2
-  PHP_GD_JISX0208
-
-else
-
- if test "$PHP_GD" != "no"; then
-  GD_MODULE_TYPE=external
-  extra_sources="gd_compat.c"
-
-dnl Header path
-  for i in include/gd include/gd2 include gd ""; do
-    test -f "$PHP_GD/$i/gd.h" && GD_INCLUDE="$PHP_GD/$i"
-  done
-
-  if test -z "$GD_INCLUDE"; then
-    AC_MSG_ERROR([Unable to find gd.h anywhere under $PHP_GD])
-  fi
-
-dnl Library path
+    PHP_GD_ZLIB
+    PHP_GD_PNG
+    PHP_GD_WEBP
+    PHP_GD_JPEG
+    PHP_GD_XPM
+    PHP_GD_FREETYPE2
+    PHP_GD_JISX0208
 
-  PHP_CHECK_LIBRARY(gd, gdSetErrorMethod,
-  [
-    PHP_ADD_LIBRARY_WITH_PATH(gd, $PHP_GD/$PHP_LIBDIR, GD_SHARED_LIBADD)
+  else
+    GD_MODULE_TYPE=external
+    extra_sources="gd_compat.c"
+    if test "$ext_shared" = "no"; then
+      PKG_CHECK_MODULES_STATIC([GDLIB], [gdlib >= 2.1.0])
+    else
+      PKG_CHECK_MODULES([GDLIB], [gdlib >= 2.1.0])
+    fi
+    PHP_EVAL_LIBLINE($GDLIB_LIBS, GD_SHARED_LIBADD)
+    PHP_EVAL_INCLINE($GDLIB_CFLAGS)
     AC_DEFINE(HAVE_LIBGD, 1, [ ])
-  ],[
-    AC_MSG_ERROR([Unable to find libgd.(a|so) >= 2.1.0 anywhere under $PHP_GD])
-  ],[
-    -L$PHP_GD/$PHP_LIBDIR
-  ])
-  PHP_GD_CHECK_VERSION
-
-  PHP_EXPAND_PATH($GD_INCLUDE, GD_INCLUDE)
- fi
+    PHP_GD_CHECK_VERSION
+  fi
 fi
 
 dnl
 dnl Common for both builtin and external GD
 dnl
 if test "$PHP_GD" != "no"; then
```

Comment:
I think this whole block can be merged with the preceding.

Reply:
Good catch. Thanks!